From ea9354565b56b08bb521d2f279ca5ef1e2f9868b Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@nxp.com>
Date: Fri, 10 Mar 2017 10:22:27 +0800
Subject: [PATCH 1692/5242] MLK-14362-2 backlight: gpio_backlight: add power
 management ops

commit  79f207562d0f563e00c70e8360fc2b7b4bc1a05f from
https://source.codeaurora.org/external/imx/linux-imx.git

Add suspend/resume power management operations for
gpio backlight.

Signed-off-by: Fancy Fang <chen.fang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/video/backlight/gpio_backlight.c |   16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/drivers/video/backlight/gpio_backlight.c b/drivers/video/backlight/gpio_backlight.c
index e470da9..a1364e7 100644
--- a/drivers/video/backlight/gpio_backlight.c
+++ b/drivers/video/backlight/gpio_backlight.c
@@ -158,10 +158,26 @@ static int gpio_backlight_probe(struct platform_device *pdev)
 MODULE_DEVICE_TABLE(of, gpio_backlight_of_match);
 #endif
 
+static int gpio_backlight_suspend(struct device *dev)
+{
+	return 0;
+}
+
+static int gpio_backlight_resume(struct device *dev)
+{
+	return 0;
+}
+
+static const struct dev_pm_ops mipi_dsi_pm_ops = {
+	.suspend = gpio_backlight_suspend,
+	.resume  = gpio_backlight_resume,
+};
+
 static struct platform_driver gpio_backlight_driver = {
 	.driver		= {
 		.name		= "gpio-backlight",
 		.of_match_table = of_match_ptr(gpio_backlight_of_match),
+		.pm = &mipi_dsi_pm_ops,
 	},
 	.probe		= gpio_backlight_probe,
 };
-- 
1.7.9.5

